src/daemon/rpmostreed-transaction-livefs.c

```diff
@@ -0,0 +1,688 @@
+{
+  static const char orig_rpmdb_path[] = "usr/share/rpm.rpmostree-orig";
+
+  /* Initial setup - load sysroot, repo, and booted deployment */
+  glnx_unref_object OstreeRepo *repo = NULL;
+  if (!ostree_sysroot_get_repo (sysroot, &repo, cancellable, error))
+    return FALSE;
+  OstreeDeployment *booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  if (!booted_deployment)
+    return glnx_throw (error, "Not currently booted into an OSTree system");
+
+  /* Overlayfs doesn't support mutation of the lowerdir. And broadly speaking,
+   * our medium term goal here is to obviate most of the unlock usage.
+   */
+  OstreeDeploymentUnlockedState unlockstate = ostree_deployment_get_unlocked (booted_deployment);
+  if (unlockstate != OSTREE_DEPLOYMENT_UNLOCKED_NONE)
+    return glnx_throw (error, "livefs is incompatible with unlocked state");
+
+  /* Find the source for /etc - either booted or pending, but down below we
+     require pending */
+  OstreeDeployment *origin_merge_deployment =
+    rpmostree_syscore_get_origin_merge_deployment (sysroot,
+                                                   ostree_deployment_get_osname (booted_deployment));
+  g_assert (origin_merge_deployment);
+  const char *booted_csum = ostree_deployment_get_csum (booted_deployment);
+  const char *target_csum = ostree_deployment_get_csum (origin_merge_deployment);
+
+  /* Require a pending deployment to use as a source - perhaps in the future we
+   * handle direct live overlays.
+   */
+  if (origin_merge_deployment == booted_deployment)
+    return glnx_throw (error, "No pending deployment");
+
+  /* Open a fd for the booted deployment */
+  g_autofree char *deployment_path = ostree_sysroot_get_deployment_dirpath (sysroot, booted_deployment);
+  glnx_fd_close int deployment_dfd = -1;
+  if (!glnx_opendirat (ostree_sysroot_get_fd (sysroot), deployment_path, TRUE,
+                       &deployment_dfd, error))
+    return FALSE;
+
+  /* Find out whether we already have a live overlay */
+  g_autofree char *live_inprogress = NULL;
+  g_autofree char *live_replaced = NULL;
+  if (!rpmostree_syscore_deployment_get_live (sysroot, booted_deployment, -1,
+                                              &live_inprogress, &live_replaced,
+                                              error))
+    return FALSE;
+  const char *resuming_overlay = NULL;
+  if (live_inprogress != NULL)
+    {
+      if (strcmp (live_inprogress, target_csum) == 0)
+        resuming_overlay = target_csum;
+    }
+  const char *replacing_overlay = NULL;
+  if (live_replaced != NULL)
+    {
+      if (strcmp (live_replaced, target_csum) == 0)
+        return glnx_throw (error, "Current overlay is already %s", target_csum);
+      replacing_overlay = live_replaced;
+    }
+
+  if (resuming_overlay)
+    g_print ("Note: Resuming interrupted overlay of %s\n", target_csum);
+  if (replacing_overlay)
+    g_print ("Note: Previous overlay: %s\n", replacing_overlay);
```

Comment:
Can we enhance this for the case where there was an interrupted overlay but towards a *different* pending csum? E.g. something like:

```c
g_print ("Note: Previous %soverlay: %s\n",
         (live_inprogress && !resuming_overlay) ? "interrupted ": "", replacing_overlay);
```

?